### ISSUE TITLE
EventSourceMapping.starting_position needs to updated to be 'optional' based on latest AWS api changes.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [terraform-providers/terraform-provider-aws#5025]
Amazon recently made it possible to have SQS events trigger Lambdas.  See: https://aws.amazon.com/blogs/aws/aws-lambda-adds-amazon-simple-queue-service-to-supported-event-sources/ for more details. 

As part of this they specifically utilize CreateSourceEventMapping to describe the mapping between SQS and Lambda: https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html

*However*, as part of this, they changed the StartingPosition message property from required to optional.  You can see that it was previous required here: https://web.archive.org/web/20171222204154/https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html

```
StartingPosition
    The position in the stream where Lambda starts reading. For valid values, see CreateEventSourceMapping in the AWS Lambda Developer Guide.
    Required: Yes
    Type: String
    Update requires: Replacement
```

But now is optional in the current set of docs:

```
StartingPosition
    The position in the stream where Lambda starts reading. For valid values, see CreateEventSourceMapping in the AWS Lambda Developer Guide.
    Required: No
    Type: String
    Update requires: Replacement
```

Unfortunately, this makes is (afaict) impossible to use terraform to configure this EventSourceMapping.  If i leave out the starting_position property, terraform complains with:

```
Plan apply failed:
Error creating Lambda event source mapping: ValidationException: 1 validation error detected: Value '' at 'startingPosition' failed to satisfy constraint: Member must satisfy enum value set: [LATEST, AT_TIMESTAMP, TRIM_HORIZON]
```

However, if i supply any of those, AWS itself errors out with:

```
Plan apply failed: 
Error creating Lambda event source mapping: InvalidParameterValueException: StartingPosition is not valid for SQS event sources.
```

**Comment by bflad** [comment]
In version 1.26.0 of the AWS provider, releasing tonight or tomorrow, the `aws_lambda_event_source_mapping` resource will make `starting_position` optional and allow `batch_size` to support default of 10 for SQS.

**Comment by bflad** [comment]
This has been released in [version 1.26.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1260-july-04-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by bflad** [comment]
@cbartram if you are having an issue, I would suggest opening a new issue following the template for all the relevant details rather than commenting on a closed issue, which could go unnoticed. 👍
